### PR TITLE
Databricks workspace service fails to deploy to East US with DNS error

### LINKED ISSUE #3515: Databricks workspace service fails to deploy with DNS error
Remember to label the issue with the appropriate area, and also assign it to yourself if you plan to work on the issue in the near future.

**Describe the bug**

Unable to deploy an instance of the default `databricks` workspace service. The error messages in the Operations logs are verbose but this is the first message:

`looking up A records for "": lookup : no such host`

**Steps to reproduce**

1. Deploy TRE 0.11
1. Deploy a new base workspace
1. Register default databricks workspace service bundle
1. Attempt to deploy the databricks service to the workspace.

### PR DESCRIPTION
**Pull request by marrobi**
Resolves #3515

## What is being addressed
- Add missing entries to databricks-udr.json
- Add notes to the docs

### CODE REVIEW

File: docs/tre-templates/workspace-services/databricks.md [hunk 1]
```diff
@@ -6,6 +6,11 @@ This service along with Azure Databricks Private Authentication Shared Service i
 
 ![Azure Databricks workspace service](../../assets/databricks_workspace_service.png)
 
+This service uses a JSON file to store the various network endpoints required by Databricks to function.
+
+If you hit networking related issues when deploying or using Databricks, please ensure this file [https://github.com/microsoft/AzureTRE/blob/main/templates/workspace_services/databricks/terraform/databricks-udr.json](https://github.com/microsoft/AzureTRE/blob/main/templates/workspace_services/databricks/terraform/databricks-udr.json) contains the approprate settings for the region you are using.
+
+The required settings for each region can be extracted from this document: [https://learn.microsoft.com/en-us/azure/databricks/resources/supported-regions](https://learn.microsoft.com/en-us/azure/databricks/resources/supported-regions).
```
Comment (SvenAelterman) [inline docs/tre-templates/workspace-services/databricks.md:13]: I would generally remove the `/en-us` from the URL to load the localized version if available and requested by the browser.

Comment (marrobi) [reply docs/tre-templates/workspace-services/databricks.md:13]:
```suggestion
The required settings for each region can be extracted from this document: [https://learn.microsoft.com/azure/databricks/resources/supported-regions](https://learn.microsoft.com/azure/databricks/resources/supported-regions).
```

**Review by SvenAelterman — CHANGES_REQUESTED** [review]
To successfully register the updated container as a new version in TRE, the version in porter.yaml should be increased.

File: CHANGELOG.md [hunk 2]
```diff
@@ -12,6 +12,7 @@ ENHANCEMENTS:
 BUG FIXES:
 * Nexus might fail to deploy due to wrong identity used in key-vault extension ([#3492](https://github.com/microsoft/AzureTRE/issues/3492))
 * Airlock notifier needs SCM basic-auth enabled to install ([#3509](https://github.com/microsoft/AzureTRE/issues/3509))
+* Databricks fails to depoly in East US ([#3515](https://github.com/microsoft/AzureTRE/issues/3515))
```
Comment (SvenAelterman) [inline CHANGELOG.md:15]: Typo: should be `deploy`

Comment (SvenAelterman) [reply CHANGELOG.md:15]: Also a typo on line 10: `natively`

Comment (marrobi) [reply CHANGELOG.md:15]:
```suggestion
* Databricks fails to deploy in East US ([#3515](https://github.com/microsoft/AzureTRE/issues/3515))
```

**Review by SvenAelterman — APPROVED** [review]

**Review by guybartal — APPROVED** [review]
I've compared the domains with https://learn.microsoft.com/en-us/azure/databricks/resources/supported-regions and it LGTM

**Pull request merged**